### PR TITLE
feat: add support for @composeDirective

### PR DESCRIPTION
**Pull request by arutkowski00**
Based on https://github.com/Gusto/apollo-federation-ruby/pull/253

### CODE REVIEW

File: .github/workflows/semantic-pr-title.yml
Comment (arutkowski00) [inline .github/workflows/semantic-pr-title.yml:1]: I renamed the file, don't know why it thinks it's a new file...

Comment (RomKadria) [reply .github/workflows/semantic-pr-title.yml:1]: cant we just delete it?

File: lib/apollo-federation/schema.rb [hunk 1]
```diff
@@ -53,20 +55,53 @@ def query(new_query_object = nil)
...
+          link_str = '  @link('
+          link_str += "url: \"#{link[:url]}\""
+          link_str += ", as: \"#{link[:as]}\"" if link[:as]
+          if link[:import]
+            imported_directives = link[:import].map { |d| "\"@#{d}\"" }.join(', ')
+            link_str += ", import: [#{imported_directives}]"
+          end
+          link_str += ')'
+          schema << link_str
+        end
```
Comment (RomKadria) [inline lib/apollo-federation/schema.rb:95]: just pointing that this code annoys me but I dont have imporement suggestion :D
bu the spaces within the string are something xD

File: lib/apollo-federation/schema.rb [hunk 2]
```diff
@@ -53,20 +55,53 @@ def query(new_query_object = nil)
...
+
+        compose_directives.each do |directive|
+          schema << "  @composeDirective(name: \"@#{directive}\")"
+        end
 
-        <<~SCHEMA
-          extend schema
-            @link(url: "https://specs.apollo.dev/federation/v#{federation_version}"#{federation_namespace}, import: [#{(IMPORTED_DIRECTIVES.map { |directive| "\"@#{directive}\"" }).join(', ')}])
+        schema << ''
+        schema << ''
```
Comment (RomKadria) [inline lib/apollo-federation/schema.rb:102]: why twice?

**Review by RomKadria — APPROVED** [review]

File: lib/apollo-federation/schema.rb (hunk 1 above)
Comment (arutkowski00) [reply lib/apollo-federation/schema.rb:95]: That's the original code made in https://github.com/Gusto/apollo-federation-ruby/pull/253 without any changes from my side

File: lib/apollo-federation/schema.rb (hunk 2 above)
Comment (arutkowski00) [reply lib/apollo-federation/schema.rb:102]: It's replaced into two newlines to make one line break between `@composeDirective` lines and the rest of the schema

**Pull request merged**